test/test_cuts.py

```diff
@@ -0,0 +1,228 @@
+            digest_values_test)
+
+# Set up some random datasets to test the cuts:
+
+# Make this reproducible
+np.random.seed(1865)
+
+# Set values for triggers:
+n_triggers = 10000
+trigger_dset = {}
+trigger_dset['snr'] = np.random.random(n_triggers) * 10
+trigger_dset['chisq'] = np.random.random(n_triggers) * 100
+trigger_dset['chisq_dof'] = np.ceil(np.random.random(n_triggers) * 100) + 2
+trigger_dset['sg_chisq'] = np.random.random(n_triggers) * 2
+
+# Set values for templates:
+template_dset = {}
+n_templates = 10000
+
+# For one of the tests, we want to use an exact value of mass1,
+# so set it manually
+mass1 = list(np.random.random(n_templates - 1) * 100) + [59]
+mass2 = list(np.random.random(n_templates - 1) * 100) + [10]
+
+# Assert mass1 > mass2
+template_dset['mass1'] = np.maximum(mass1, mass2)
+template_dset['mass2'] = np.minimum(mass1, mass2)
+
+# Spins must be in the range -1, 1, (but not exactly 1):
+def make_random_spins(n_templates):
+    spins = np.random.normal(size=n_templates, scale=0.4)
+    spins = np.maximum(spins, -0.998)
+    spins = np.minimum(spins, 0.998)
+    return spins
+
+template_dset['spin1z'] = make_random_spins(n_templates)
+template_dset['spin2z'] = make_random_spins(n_templates)
+
+template_dset['template_duration'] = np.random.random(n_templates) * 100
+
+
+class CutsTest(unittest.TestCase):
+    def setUp(self):
+        # Set up some things we will want to use in the tests:
+        self.maxDiff = None
+
+# Set up values to be tested:
+test_cut_output = []
+
+
+# Lower limits on a trigger parameter we can read directly
+def snr_lower_test(trigs, trig_idx, **kwargs):
+    return all(trigs['snr'][trig_idx] > 4)
+
+
+test_cut_output.append((['--trigger-cuts', 'snr:4:lower'],
+                        snr_lower_test,
+                        'snr_cut_lower'))
+
+
+# Upper limits on a derived trigger parameter
+def newsnr_sgveto_upper_test(trigs, trig_idx, **kwargs):
+    rwsnr = ranking.get_newsnr_sgveto(trigs)
+    return all(rwsnr[trig_idx] < 10)
+
+test_cut_output.append((['--trigger-cuts', 'newsnr_sgveto:10:upper'],
+                        newsnr_sgveto_upper_test,
+                        'nsnr_sgveto_cut_upper'))
+
+# Lower limit on directly-read template bank parameter
+def template_duration_test(temps, temp_idx, **kwargs):
+    return all(temps['template_duration'][temp_idx] > 10)
+
+test_cut_output.append((['--template-cuts',
+                         'template_duration:10:lower'],
+                        template_duration_test,
+                        'template_duration_lower'))
+
+# make sure the "or equal to" is working properly
+def mass1_ge_test(temps, temp_idx, **kwargs):
+    return any(temps['mass1'][temp_idx] == 59)
+
+test_cut_output.append((['--template-cuts',
+                         'mass1:59:lower_inc'],
+                        mass1_ge_test,
+                        'mass1_orequal'))
+
+# Upper and lower limits on a derived parameter from the bank
```

Comment:
I've added some unit tests for the module

The important test for the point of this PR/problem is this line. Previously, only one of these changes would have been made